OpenEdXMobile/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -188,28 +189,27 @@ String getUserEnrolledCoursesURL(@NonNull String username) {
 
         // if we don't have a json yet, get it from userService
         if (json == null) {
-            Response<ResponseBody> response = userService.getUserEnrolledCourses(username, org).execute();
-            if (response.isSuccessful()) {
-                json = userService.getUserEnrolledCourses(username, org).execute().body().string();
+            try {
+                json = executeStrict(userService.getUserEnrolledCourses(username, org)).string();
                 // cache result
                 try {
                     cache.put(cacheKey, json);
                 } catch (IOException e) {
                     logger.debug(e.toString());
                 }
-            } else {
+            } catch (HttpResponseStatusException connectivityException) {
```

Comment:
nit: This won't necessarily be a connectivity exception, it maybe a server based exception as well. `responseException` might be a better name for it.